pcsx2/GS/Renderers/SW/GSDrawScanline.cpp

```diff
@@ -545,11 +545,14 @@ void GSDrawScanline::DrawScanline(int pixels, int left, int top, const GSVertexS
 
 				if (sel.prim != GS_SPRITE_CLASS)
 				{
+					// Need to handle when the float converts incorrectly
 					GSVector8 z = GSVector8::broadcast32(&scan.p.z) + zo;
-
 					if (sel.zoverflow)
 					{
-						zs = (GSVector8i(z * 0.5f) << 1) | (GSVector8i(z) & GSVector8i::x00000001());
+						z *= 0.5f;
+						zs = GSVector8i(z, true);
+						zs = zs.min_u32(GSVector8i::x7fffffff());
+						zs = zs.sll32(1) | 1;
```

Comment:
I don't think this is correct
Let's say z is `10.f`
```
z *= 0.5f; // z = 5.f
zs = GSVector8i(z, true); // zs = 5
zs = zs.min_u32(GSVector8i::x7fffffff()); // zs = 5
zs = zs.sll32(1) | 1; // zs = 11(!)
```

Reply:
It's not correct, you're right, but before it cut off the lower 255, so if it was 255 it became 0 :P so although this is still wrong, it's less wrong, the ideal solution is to convert it all to integer, since it's impossible (without just using intrinsics) to convert to unsigned 32bit from float, which makes things more difficult.

In the case of 007, it draws triangles with UINT_MAX as the 32bit depth, then draws a sprite over the top, but it causes bad values to be left in the depth and it misses drawing half of the sprite, so the value from the triangle draw needs to be UINT_MAX and the only way to do that when we're dividing by 2 is to or in the extra 1.

But yeah, it's wrong, but it's less wrong than it was.